packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -35,6 +40,72 @@ const Indicator = () => {
   const trackedInitialLoad = useRef(false)
   const { track } = useTrackEvent()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+    try {
+      const resp = await fetch(url)
+      const data = await resp.text()
+
+      if (data === `{}`) {
+        const err = new Error(`Not Found`)
+        err.status = 404
+        throw err
+      }
+
+      return { data, errorMessage: null }
+    } catch (e) {
+      if (e.status === 404) {
```

Comment:
Can you maybe add a comment here explaining what's happening.

If you fetch and the page is not there, it returns an `{}`?